### PR TITLE
Change default optimization_level

### LINKED ISSUE #643: Different defaults for error mitigation/suppression for noiseless backends
**What is the expected feature or enhancement?**

Current default options for the primitives are:

```
optimization_level = 3
resilience_level = 1
```

These options add a processing overhead for error suppression and error mitigation. When using noiseless devices, the overhead doesn't bring any additional improvements. We should consider using different defaults when using noiseless devices (i.e. simulator without noise model)

**Acceptance criteria**

### PR DESCRIPTION
**Pull request by merav-aharoni**
### Summary
Change default `optimization_level` to None. If the backend is a simulator without noise, `optimization_level` will be set to 1. Otherwise it will be set to 3.

### Details and comments
Fixes #643

### CODE REVIEW

**Comment by merav-aharoni** [comment]
@kt474 , @jyu00 - I implemented as we discussed. 
There is one problem - in `test_ibm_primitives.py` several tests have an assertion `self.assertDictEqual(inst.options.__dict__, asdict(Options()))`. 
This assertion now fails because the default `optimization_level` is None , but in `inst.options` it is 1 (or 3). 
Any suggestions on how to resolve? One options is to check the fields one at a time and skip `optimization_level`.

**Comment by kt474** [comment]
Looks great so far! 

What you suggested would work for the test cases - since there are only 3 tests that are failing you could something like the following: 

`test_run_multiple_different_options`, `test_run_overwrite_options` - just update the assert statement 
```python 
self.assertDictEqual(inst.options.__dict__, asdict(Options(optimization_level=1)))
```

`test_dict_options` same thing for this one 
```python
expected = asdict(Options(optimization_level=1))
```

**Comment by merav-aharoni** [comment]
Thank you, @kt474 . Good idea! I will complete this and then I'll ask for your review.

**Comment by merav-aharoni** [comment]
@kt474 - I think this PR is ready for review now.

File: test/integration/test_options.py [hunk 1]
```diff
@@ -68,6 +68,7 @@ def test_noise_model(self, service):
     @run_integration_test
     def test_simulator_transpile(self, service):
         """Test simulator transpile options."""
+        print("servie = " + str(dir(service)))
```
Comment (kt474) [inline test/integration/test_options.py:71]: This can be removed

File: releasenotes/notes/default-optimization-level-3197ff933dfcfa4f.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+
```
Comment (kt474) [inline releasenotes/notes/default-optimization-level-3197ff933dfcfa4f.yaml:2]:
```suggestion
```
The extra space may mess with the docs formatting 
This release note can be categorized as "upgrade"

File: test/integration/test_options.py [hunk 3]
```diff
@@ -98,3 +99,28 @@ def test_simulator_transpile(self, service):
                         job2.result()
                     # TODO: Re-enable when ntc-1651 is fixed
                     # self.assertIn("TranspilerError", err.exception.message)
+
+    @run_integration_test
+    def test_optimization_level(self, service):
+        """Test various definitions for optimization_level."""
+
+        backend = service.backends(simulator=True)[0]
```
Comment (kt474) [inline test/integration/test_options.py:107]: nit: ```service.get_backend('ibmq_qasm_simulator")```

**Review by kt474 — COMMENTED** [review]
Looks pretty good to me! Some minor comments - also I think the original issue #643 mentioned different default values for both `optimization_level` and `resilience_level`

File: test/integration/test_options.py (hunk 1 above)
Comment (merav-aharoni) [reply test/integration/test_options.py:71]: Of course...

**Comment by merav-aharoni** [comment]
Regarding `resilience_level`, what I understood from @jyu00 was that I should only take care of the `optimization_level`. But I'll be happy to do something similar for `resilience_level`. What do you think, @jyu00 , @kt474 ? Should the conditions be the same (0 for simulation with no noise model, 1 otherwise)?

File: test/integration/test_options.py (hunk 3 above)
Comment (merav-aharoni) [reply test/integration/test_options.py:107]: No problem, but this line appears in several tests in this file. Shall I change them all to be consistent?

Comment (kt474) [reply test/integration/test_options.py:107]: ah I didn't notice that - you can leave it as is then

**Comment by jyu00** [comment]
> Regarding `resilience_level`, what I understood from @jyu00 was that I should only take care of the `optimization_level`. But I'll be happy to do something similar for `resilience_level`. What do you think, @jyu00 , @kt474 ? Should the conditions be the same (0 for simulation with no noise model, 1 otherwise)?

Sorry if I wasn't clear before. We definitely should do `resilience_level` as well, since the overhead is even bigger than `optimization_level`

**Comment by merav-aharoni** [comment]
Thanks, @jyu00 , my misunderstanding. I will add to the PR.

**Comment by merav-aharoni** [comment]
Hi @kt474 , I addressed your comments and added support for the `resilience_level`. Can you please review again?

**Review by kt474 — APPROVED** [review]
LGTM!

File: qiskit_ibm_runtime/options/options.py [hunk 4]
```diff
@@ -81,7 +81,11 @@ class Options:
             :class:`SimulatorOptions` for all available options.
     """
 
-    optimization_level: int = 3
+    # Defaults for optimization_level and for resilience_level will be assigned
+    # in Sampler/Estimator
+    _DEFAULT_OPTIMIZATION_LEVEL = 3
+    _DEFAULT_RESILIENCE_LEVEL = 1
+    optimization_level: Optional[int] = None
     resilience_level: int = 1
```
Comment (kt474) [inline qiskit_ibm_runtime/options/options.py:89]: `resilience_level` should also be optional right?

File: qiskit_ibm_runtime/sampler.py [hunk 5]
```diff
@@ -169,6 +169,18 @@ def __init__(
...
+                not hasattr(_options.simulator, "noise_model")
+                or asdict(_options.simulator)["noise_model"] is None
+            ):
+                _options.optimization_level = 1
+                _options.resilience_level = 0
+            else:
+                _options.optimization_level = Options._DEFAULT_OPTIMIZATION_LEVEL
+                _options.resilience_level = Options._DEFAULT_RESILIENCE_LEVEL
+
         self._options: dict = asdict(_options)
```
Comment (kt474) [inline qiskit_ibm_runtime/sampler.py:184]: Sorry, didn't see this initially - users can set optimization level or resilience level separately so the logic would have to something like: 

```python
if _options.optimization_level is None:
      # set values appropriately 
            
if _options.resilience_level is None:
      # set values appropriately 
            
```

File: qiskit_ibm_runtime/options/options.py (hunk 4 above)
Comment (merav-aharoni) [reply qiskit_ibm_runtime/options/options.py:89]: Certainly. It was actually this way in my clone, but I guess I forgot to commit.

File: qiskit_ibm_runtime/sampler.py (hunk 5 above)
Comment (merav-aharoni) [reply qiskit_ibm_runtime/sampler.py:184]: Of course you are right - I don't know how I missed this. Will fix.

**Comment by merav-aharoni** [comment]
@kt474 - I made the fixes you pointed out and also improved one of the tests. Please have another (hopefully final) review.

**Pull request merged**